robottelo/libvirt_discovery.py

```diff
@@ -118,11 +125,14 @@ def create(self):
             '--os-variant=rhel7',
             '--disk path={image_name},size=8',
             '--noautoconsole',
+            '--graphics spice,listen=0.0.0.0',
         ]
 
         if not self.boot_iso:
             # Required for PXE-based host discovery
-            command_args.append('--pxe')
+            # command_args.append('--pxe')
+            command_args.append('--boot network')
```

Comment:
Could you please add some explanation to this change? Is it backwards compatible?

Reply:
Sure, here's the description from `virt-install` manpage:

```
--pxe
           Install from PXE. This just tells the VM to boot off the network for the first boot.
```
```
--boot BOOTOPTS
           Optionally specify the post-install VM boot configuration. This option allows specifying a boot device
           order, permanently booting off kernel/initrd with option kernel arguments, and enabling a BIOS boot menu
           (requires libvirt 0.8.3 or later)

           --boot cdrom,fd,hd,network
               Set the boot device priority as first cdrom, first floppy, first harddisk, network PXE boot.

```

tl;dr - the usage of `--pxe` has been wrong since the beginning - it is used for 1-time boot and such machine won't survive reboots.

the `--boot network` on the other hand properly sets the persistent boot device (order) so the VM uses it even after reboot - since we also test VM behavior after reboot, this is a necessity.

Btw, i should probably remove the commented line